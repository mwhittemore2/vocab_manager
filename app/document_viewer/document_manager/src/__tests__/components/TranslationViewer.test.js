import React from "react"
import { shallow } from "enzyme"
import toJSON from 'enzyme-to-json'
import { BaseFormMatch, Definitions, DerivedFormMatch,
         TranslationViewer, ViewMatches } from '../../components/ui/TranslationViewer'
import C from '../../constants'

describe("Definitions", () => {
    it("single definition", () => {
        let match = {
                definitions: ["book"]
            }
        let page = 1
        let txt = toJSON(
                      shallow(<Definitions match={match}
                                           page={page}/>)
                  )
        expect(txt).toMatchSnapshot()
    })

    it("multiple definitions", () => {
        let match = {
                definitions: ["book", "novel", "writing"]
            }
        let page = 1
        let txt = toJSON(
                      shallow(<Definitions match={match}
                                           page={page}/>
                      )
                  )
        expect(txt).toMatchSnapshot()
    })

    it("next page", () => {
        let match = {
                definitions: ["lamp", "light"]
            }
        let page = 2
        let txt = toJSON(
                      shallow(<Definitions match={match}
                                           page={page}/>
                      )
                  )
        expect(txt).toMatchSnapshot()
    })
})

describe("BaseFormMatch", () => {
    it("single definition", () => {
        let match = {
                text: "Welt",
                pos: "Noun",
                definitions: ["world"]
            }
        let page = 1
        let txt = toJSON(
                      shallow(<BaseFormMatch match={match}
                                             page={page}/>
                      )
                  )
        expect(txt).toMatchSnapshot()
    })

    it("multiple definitions", () => {
        let match = {
                text: "Welt",
                pos: "Noun",
                definitions: ["world", "earth"]
            }
        let page = 1
        let txt = toJSON(
                      shallow(<BaseFormMatch match={match}
                                             page={page}/>
                      )
                  )
        expect(txt).toMatchSnapshot()
    })
})

describe("DerivedFormMatch", () => {
    it("single definition", () => {
        let match = {
                text: "Welten",
                definitions: ["worlds"],
                see_also:{
                    text: "Welt",
                    pos: "Noun"
                }
            }
        let page = 1
        let txt = toJSON(
                      shallow(<DerivedFormMatch match={match}
                                                page={page}/>
                      )
                  )
        expect(txt).toMatchSnapshot()
    })

    it("multiple definitions", () => {
        let match = {
                text: "Welten",
                definitions: ["worlds", "earths"],
                see_also: {
                    text: "Welt",
                    pos: "Noun"
                }
            }
        let page = 1
        let txt = toJSON(
                      shallow(<DerivedFormMatch match={match}
                                                page={page}/>
                      )
                  )
        expect(txt).toMatchSnapshot()
    })
})

describe("ViewMatches", () => {
    it("single match", () => {
        let translations = {
                currPage: 1,
                matches: [
                    {
                        text: "Welt",
                        pos: "Noun",
                        definitions: ["world", "earth"]
                    }
                ]
            }
        let txt = toJSON(
                      shallow(<ViewMatches translations={translations}/>)
                  )
        expect(txt).toMatchSnapshot()
    })

    it("multiple matches", () => {
        let translations = {
            currPage: 1,
            matches: [
                {
                    text: "Welt",
                    pos: "Noun",
                    definitions: ["world", "earth"]
                },
                {
                    text: "Welten",
                    definitions: ["worlds", "earths"],
                    see_also: {
                        text: "Welt",
                        pos: "Noun"
                    }
                }
            ]
        }
        let txt = toJSON(
                      shallow(<ViewMatches translations={translations}/>)
                  )
        expect(txt).toMatchSnapshot()
    })
})

describe("TranslationViewer" , () => {
    let closeViewer
    let cursor
    let translations
    beforeEach(() => {
        closeViewer = jest.fn()
        cursor = jest.fn()
        translations = {
            currPage: 1,
            matches: [
                {
                    text: "Welt",
                    pos: "Noun",
                    definitions: ["world", "earth"]
                }
            ]
        }
    })

    afterAll(() => {
        jest.resetAllMocks()
    })

    it("view translations", () => {
        let interaction = C.DOCUMENT_VIEWER
        let option = C.TRANSLATION_VIEWER
        let loaded = {}
        loaded[C.TRANSLATION_VIEWER] = true
        let txt = toJSON(
                      shallow(<TranslationViewer closeViewer={closeViewer}
                                                 cursor={cursor}
                                                 interaction={interaction}
                                                 loaded={loaded}
                                                 option={option}
                                                 translations={translations}/>
                      )
                  )
        expect(txt).toMatchSnapshot()
    })

    it("loading", () => {
        let interaction = C.DOCUMENT_VIEWER
        let option = C.TRANSLATION_VIEWER
        let loaded = {}
        loaded[C.TRANSLATION_VIEWER] = false
        let txt = toJSON(
                      shallow(<TranslationViewer closeViewer={closeViewer}
                                                 cursor={cursor}
                                                 interaction={interaction}
                                                 loaded={loaded}
                                                 option={option}
                                                 translations={translations}/>
                      )  
                  )
        expect(txt).toMatchSnapshot()
    })

    it("not in document viewer", () =>{
        let interaction = C.DOCUMENT_SELECTOR
        let option = C.TRANSLATION_VIEWER
        let loaded = {}
        loaded[C.TRANSLATION_VIEWER] = true
        let txt = toJSON(
                      shallow(<TranslationViewer closeViewer={closeViewer}
                                                 cursor={cursor}
                                                 interaction={interaction}
                                                 loaded={loaded}
                                                 option={option}
                                                 translations={translations}/>
                      ).html()    
                  )
        expect(txt).toMatchSnapshot()
    })
})

