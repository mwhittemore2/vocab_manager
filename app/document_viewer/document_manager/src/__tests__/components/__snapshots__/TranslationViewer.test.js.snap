// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseFormMatch multiple definitions 1`] = `
<div>
  Text: 
  Welt
  <br />
  POS: 
  Noun
  <br />
  Definitions:
  <Definitions
    match={
      Object {
        "definitions": Array [
          "world",
          "earth",
        ],
        "pos": "Noun",
        "text": "Welt",
      }
    }
    page={1}
  />
</div>
`;

exports[`BaseFormMatch single definition 1`] = `
<div>
  Text: 
  Welt
  <br />
  POS: 
  Noun
  <br />
  Definitions:
  <Definitions
    match={
      Object {
        "definitions": Array [
          "world",
        ],
        "pos": "Noun",
        "text": "Welt",
      }
    }
    page={1}
  />
</div>
`;

exports[`Definitions multiple definitions 1`] = `
<ol>
  <li
    key="tr-def-1-0"
  >
    book
  </li>
  <li
    key="tr-def-1-1"
  >
    novel
  </li>
  <li
    key="tr-def-1-2"
  >
    writing
  </li>
</ol>
`;

exports[`Definitions next page 1`] = `
<ol>
  <li
    key="tr-def-2-0"
  >
    lamp
  </li>
  <li
    key="tr-def-2-1"
  >
    light
  </li>
</ol>
`;

exports[`Definitions single definition 1`] = `
<ol>
  <li
    key="tr-def-1-0"
  >
    book
  </li>
</ol>
`;

exports[`DerivedFormMatch multiple definitions 1`] = `
<div>
  Text: 
  Welten
  <br />
  Definitions:
  <Definitions
    match={
      Object {
        "definitions": Array [
          "worlds",
          "earths",
        ],
        "see_also": Object {
          "pos": "Noun",
          "text": "Welt",
        },
        "text": "Welten",
      }
    }
    page={1}
  />
  <br />
  See Also: 
  Welt
   (
  Noun
  )
  <br />
</div>
`;

exports[`DerivedFormMatch single definition 1`] = `
<div>
  Text: 
  Welten
  <br />
  Definitions:
  <Definitions
    match={
      Object {
        "definitions": Array [
          "worlds",
        ],
        "see_also": Object {
          "pos": "Noun",
          "text": "Welt",
        },
        "text": "Welten",
      }
    }
    page={1}
  />
  <br />
  See Also: 
  Welt
   (
  Noun
  )
  <br />
</div>
`;

exports[`TranslationViewer loading 1`] = `
<div
  id="translation-viewer"
>
  <Loading />
</div>
`;

exports[`TranslationViewer not in document viewer 1`] = `undefined`;

exports[`TranslationViewer view translations 1`] = `
<div
  id="translation-viewer"
>
  <button>
    Close
  </button>
  <ViewMatches
    translations={
      Object {
        "currPage": 1,
        "matches": Array [
          Object {
            "definitions": Array [
              "world",
              "earth",
            ],
            "pos": "Noun",
            "text": "Welt",
          },
        ],
      }
    }
  />
  <Navigator
    cursor={[MockFunction]}
    results={
      Object {
        "currPage": 1,
        "matches": Array [
          Object {
            "definitions": Array [
              "world",
              "earth",
            ],
            "pos": "Noun",
            "text": "Welt",
          },
        ],
      }
    }
    viewer="translation-viewer"
  />
</div>
`;

exports[`ViewMatches multiple matches 1`] = `
<div
  id="translation-matches"
>
  <BaseFormMatch
    key="tr-match-1-0"
    match={
      Object {
        "definitions": Array [
          "world",
          "earth",
        ],
        "pos": "Noun",
        "text": "Welt",
      }
    }
    page={1}
  />
  <DerivedFormMatch
    key="tr-match-1-1"
    match={
      Object {
        "definitions": Array [
          "worlds",
          "earths",
        ],
        "see_also": Object {
          "pos": "Noun",
          "text": "Welt",
        },
        "text": "Welten",
      }
    }
    page={1}
  />
</div>
`;

exports[`ViewMatches single match 1`] = `
<div
  id="translation-matches"
>
  <BaseFormMatch
    key="tr-match-1-0"
    match={
      Object {
        "definitions": Array [
          "world",
          "earth",
        ],
        "pos": "Noun",
        "text": "Welt",
      }
    }
    page={1}
  />
</div>
`;
